pkg/kapp/crdupgradesafety/change_validator.go

```diff
@@ -71,6 +71,48 @@ func EnumChangeValidation(diff FieldDiff) (bool, error) {
 	return handled(), nil
 }
 
+// RequiredFieldChangeValidation adds a validation check to ensure that
+// existing required fields can be marked as optional in a CRD schema:
+// - No new values are added as required that did not previously have
+// any required fields present
+// - Existing values are removed from the required field
+// This function returns:
+// - A boolean representation of whether or not the change
+// has been fully handled (i.e. the only change was to required field values)
+// - An error if either of the above criteria are not met
+func RequiredFieldChangeValidation(diff FieldDiff) (bool, error) {
+	handled := func() bool {
+		diff.Old.Required = []string{}
+		diff.New.Required = []string{}
+		return reflect.DeepEqual(diff.Old, diff.New)
+	}
+
+	if len(diff.Old.Required) == 0 && len(diff.New.Required) > 0 {
+		return handled(), fmt.Errorf("new values added as required when previously no required fields existed")
```

Comment:
Nit: Update the error message to include the set of fields added as required

Reply:
Addressed in latest [commit](https://github.com/carvel-dev/kapp/pull/933/commits/e77c330f99472ca109067c784a49c7f9ef039b86)